tests/saw/spec/handshake/handshake_io_lowlevel.saw

```diff
@@ -64,21 +61,61 @@ let config_cca_type config = (crucible_field config "client_cert_auth_type");
 let ocsp_status_size cert_and_key =
     crucible_field (crucible_field (cert_and_key) "ocsp_status") "size";
 
-// conn->config->use_tickets and conn->config->quic_enabled are the fields we
-// care about, but they are in a bitfield. We currently handle this by
-// initializing the entire bitfield with a symbolic value and imposing a
-// precondition that bit 3 (i.e., use_tickets) must be equal to 0.
-// Bit 9 (i.e., quic_enabled), on the other hand, is allowed to be either
-// 0 or 1.
+// At the moment, we care about the following fields that are located in
+// bitfields:
+//
+// * conn->corked_io
+// * conn->client_session_resumed
+// * conn->quic_enabled
+// * conn->config->use_tickets
+// * conn->config->quic_enabled
+//
+// We currently handle this by initializing the entirety of each bitfield with
+// a symbolic value and imposing appropriate preconditions on them. See the
+// comments in setup_connection_common next to conn_bitfield and
+// config_bitfield for more details about these preconditions.
 
-// It's also noteworthy that we use the crucible_elem here, which will
+// It's also noteworthy that we use crucible_elem here, which will
 // cause problems if the fields are reordered. This is reasonably unlikely
 // because it makes sense to have the bitfield at the start, but it is a bit
 // fragile nonetheless. If https://github.com/GaloisInc/saw-script/issues/1461
 // were implemented, we could reference the names of bitfields instead of their
 // offsets, which would avoid this issue.
+let conn_bitfield pconn = (crucible_elem pconn 0);
 let config_bitfield config = (crucible_elem config 0);
 
+// These represent the amount of memory that LLVM will use to represent the
+// bitfields in s2n_connection and s2n_config, respectively. The size is
+// determined by counting the number of fields in the bitfield and rounding
+// up to the nearest power of 2.
+let conn_bitfield_size = 32;
+let config_bitfield_size = 16;
+
+let {{
+// The position (0-indexed) of each field of interest in
+// s2n_connection's bitfield.
+conn_corked_io_index : [conn_bitfield_size]
+conn_corked_io_index = 0
+
+conn_client_session_resumed_index : [conn_bitfield_size]
+conn_client_session_resumed_index = 1
+
+conn_quic_enabled_index : [conn_bitfield_size]
+conn_quic_enabled_index = 16
+
+// The position (0-indexed) of each field of interest in
+// s2n_config's bitfield.
+config_use_tickets_index : [config_bitfield_size]
+config_use_tickets_index = 2
+
+config_quic_enabled_index : [config_bitfield_size]
+config_quic_enabled_index = 8
```

Comment:
While we're reordering fields, can we move the flags tracked by SAW to the front of the C bitfield (to reduce the chances that we insert another flag above them and change their index) and put a comment in the C code warning that their order matters for SAW?

Reply:
Good idea. I've pushed a commit which does this.